akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -470,13 +470,18 @@ object Source {
    *
    * The stream can be completed successfully by sending the actor reference a message that is matched by
    * `completionMatcher` in which case already buffered elements will be signaled before signaling
-   * completion, or by sending [[akka.actor.PoisonPill]] in which case completion will be signaled immediately.
+   * completion.
    *
    * The stream can be completed with failure by sending a message that is matched by `failureMatcher`. The extracted
    * [[Throwable]] will be used to fail the stream. In case the Actor is still draining its internal buffer (after having received
    * a message matched by `completionMatcher`) before signaling completion and it receives a message matched by `failureMatcher`,
    * the failure will be signaled downstream immediately (instead of the completion signal).
    *
+   * Note that terminating the actor without first completing it, either with a success or a
+   * failure, will prevent the actor triggering downstream completion and the stream will continue
+   * to run even though the source actor is dead. Therefore you should **not** attempt to
+   * manually terminate the actor such as with a [[akka.actor.PoisonPill]].
+   *
```

Comment:
also corresponding javadsl/Source

Reply:
There is also in the reference documentation: https://doc.akka.io/docs/akka/current/stream/stream-integrations.html#source-actorref and https://doc.akka.io/docs/akka/current/stream/operators/Source/actorRef.html

Reply:
Updated the docs in those places. I also made a couple of grammar fixes and removed some trailing whitespace while I was there.